src/libraries/Common/tests/Tests/System/MarvinTests.cs

```diff
@@ -27,9 +27,13 @@ public class MarvinTests
         [MemberData(nameof(TestDataAndExpectedHashes))]
         public void ComputeHash_Success(ulong seed, string testDataString, ulong expectedHash)
         {
-            var testDataSpan = new Span<byte>(testDataString.HexToByteArray());
-            long hash = Marvin.ComputeHash(testDataSpan, seed);
-            Assert.Equal((long)expectedHash, hash);
+            // The test vectors are only valid for little-endian architectures
+            if (BitConverter.IsLittleEndian)
```

Comment:
instead of just skipping the test you could change the expected hashes in TestDataAndExpectedHashes based on the endianness, e.g. something like this:

```csharp
  public static object[][] TestDataAndExpectedHashes =
  {
      new object[] { Seed1, TestDataString0Byte, BitConverter.IsLittleEndian ? 0x30ED35C100CD3C7DLU : /* big endian value */ },
      ...
  }
```

Reply:
Well, if I knew what the expected hashes were ...   I didn't find any document specifying the Marvin algorithm, so I don't know where the test vectors for little-endian come from.   I could of course just use whatever the algorithm currently produces.

This brings me to the more fundamental question: should the algorithm actually have different results on big- vs. little-endian or should they be the same?   I assumed they should be different because existing code in ./System.Private.CoreLib/src/System/Marvin.cs had already special-cased handling for big-endian systems, resulting in the different output that we currently see.  This patch just brings the two other implementations in line with that.

Reply:
hmm that's a good question. I found https://www.larc.usp.br/users/mjunior/www/files/en/Marvin+LetterSoup.pdf but I'll need to defer to @GrabYourPitchforks about the details as I'm not familiar with it myself :)